module/data/shared/damage-field.mjs

```diff
@@ -0,0 +1,126 @@
+import FormulaField from "../fields/formula-field.mjs";
+
+const { BooleanField, EmbeddedDataField, NumberField, SchemaField, SetField, StringField } = foundry.data.fields;
+
+/**
+ * Field for storing damage data.
+ */
+export default class DamageField extends EmbeddedDataField {
+  constructor(options) {
+    super(DamageData, options);
+  }
+}
+
+/* -------------------------------------------- */
+
+/**
+ * Data model that stores information on a single damage part.
+ *
+ * @property {number} number           Number of dice to roll.
+ * @property {number} denomination     Die denomination to roll.
+ * @property {string} bonus            Bonus added to the damage.
+ * @property {Set<string>} types       One or more damage types. If multiple are selected, then the user will be able to
```

Comment:
I am ok with supporting multiple damage types here, but I do think we should really only show the single drop-down on the sheet. Multiple damage types will never be used unless the table is using some homebrew or 3rd-party content. You can leave that for me though in the UI pass if you want, unless you feel strongly otherwise.

Reply:
This provides a selection of possible damage types, for a spell like Chromatic Orb or the upcast damage on Flame Strike.

Reply:
Fair point about *chromatic orb*, but how would this work for *flame strike*? You add the fire and radiant damage parts with no scaling, then you add a 3rd fire/radiant damage part of 0d6 with scaling? I guess maybe I answered my own question.